### PR TITLE
Feature: Town and City Zone Configuration Seperated

### PR DESCRIPTION
**Pull request by reldred**
Adds additional zone configuration for cities, renames existing zone configuration to towns.

I'm a little unsure if the syntax/styling of my changes in town_cmd.cpp are good enough, but it's survived some abuse in scenario editor and behaves nicely.

### CODE REVIEW

File: src/town_cmd.cpp [hunk 1]
```diff
@@ -1978,7 +1978,14 @@ void UpdateTownRadius(Town *t)
 		{121, 81,  0, 49, 36}, // 88
 	};
 
-	if (_settings_game.economy.town_zone_calc_mode) {
+	if (_settings_game.economy.town_zone_calc_mode, t->larger_town) {
```
Comment (JGRennison) [inline src/town_cmd.cpp:1981]: Presumably this should be the && operator?
As written this uses the custom values for cities even when the setting is off.

**Comment by JGRennison** [comment]
Existing savegames which use custom town zone values expect those to be applied to both small and large towns, however when loaded with this applied cities will revert to using the default values.
The city zone values should be copied from the town zone values in that case. Probably the easiest way to do that is add a feature tag to the savegame, I can take care of this bit.

File: src/town_cmd.cpp (hunk 1 above)
Comment (reldred) [reply src/town_cmd.cpp:1981]: That's a brain fart and a half, will fix.

Comment (reldred) [reply src/town_cmd.cpp:1981]:
```suggestion
	if (_settings_game.economy.town_zone_calc_mode && t->larger_town) {
```

**Comment by JGRennison** [comment]
The branch seems to have got a bit muddled, as it has both your original commit and the rebased version.
Could be a chance to practice your git magic :)

**Comment by reldred** [comment]
> 
> 
> The branch seems to have got a bit muddled, as it has both your original commit and the rebased version.
> Could be a chance to practice your git magic :)

oh god... what did I do??? 😂 I'm going to say attempting Git Magic is why I'm in this problem in the first place 😂

**Pull request merged**